app/display/representation-javafx/src/main/java/org/csstudio/display/builder/representation/javafx/WidgetInfoDialog.java

```diff
@@ -134,17 +154,104 @@ public WidgetInfoDialog(final Widget widget, final Collection<NameStateValue> pv
+                .append(horizontalRuler).append(System.lineSeparator());
+        pvs.stream().sorted(Comparator.comparing(pv -> pv.name)).forEach(pv -> {
+            buffer.append(pv.name).append(itemSeparator).append(pv.state).append(itemSeparator).append(getPVValue(pv.value)).append(System.lineSeparator());
+        });
+        buffer.append(System.lineSeparator());
+
+        if (widget instanceof DisplayModel){
+            buffer.append("Widget Counts (widget type, count)").append(System.lineSeparator())
+                .append(horizontalRuler).append(System.lineSeparator());
+            stats.getTypes().entrySet().stream().forEach(entry -> {
+                buffer.append(entry.getKey()).append(itemSeparator).append(entry.getValue().get()).append(System.lineSeparator());
+            });
+            buffer.append(Messages.WidgetInfoDialog_Total).append(itemSeparator).append(stats.getTotal()).append(System.lineSeparator());
+            buffer.append(Messages.RulesDialog_Title).append(itemSeparator).append(stats.getRules()).append(System.lineSeparator());
+            buffer.append(Messages.ScriptsDialog_Title).append(itemSeparator).append(stats.getScripts()).append(System.lineSeparator());
+            buffer.append(System.lineSeparator());
+        }
+
+        buffer.append("Macros (name, value)").append(System.lineSeparator())
+                .append(horizontalRuler).append(System.lineSeparator());
+        macros.forEach((name, value) -> {
+            buffer.append(name).append(itemSeparator).append(value).append(System.lineSeparator());
+        });
+        buffer.append(System.lineSeparator());
+
+        buffer.append("Properties (category, property, name, value)").append(System.lineSeparator())
+                .append(horizontalRuler).append(System.lineSeparator());
+        widget.getProperties().stream().forEach(widgetProperty -> {
+            buffer.append(widgetProperty.getCategory().getDescription()).append(itemSeparator);
+            buffer.append(widgetProperty.getDescription()).append(itemSeparator);
+            buffer.append(widgetProperty.getName()).append(itemSeparator);
+            buffer.append(widgetProperty.getValue()).append(System.lineSeparator());
+        });
+        buffer.append(System.lineSeparator());
+
+        try (FileOutputStream fileOutputStream = new FileOutputStream(file)){
+            fileOutputStream.write(buffer.toString().getBytes());
+            fileOutputStream.flush();
+            Alert alert = new Alert(AlertType.INFORMATION);
+            alert.setTitle("Information Dialog");
+            alert.setHeaderText(null);
+            alert.setContentText(MessageFormat.format(Messages.ExportDone, file.getAbsolutePath()));
+            alert.showAndWait();
+        } catch (IOException e) {
+            Logger.getLogger(getClass().getName()).log(Level.WARNING, "Failed to export widget info", e);
+            Alert alert = new Alert(AlertType.ERROR);
```

Comment:
To display an error with exception, consider using `ExceptionDetailsErrorDialog` which shows the error message and allows users to expand the full exception, with option to copy the text out.

Reply:
Sure, that would make it easier to debug. I keep forgetting about `ExceptionDetailsErrorDialog`...

Reply:
Another good practice for any dialog is using `DialogHelper.positionDialog(...` to locate it relative to for example the button that triggers a dialog, because otherwise users with more than one monitor often get it on the 'other' monitor.